src/resolve/resolved-dns-packet.c

```diff
@@ -2570,6 +2570,67 @@ bool dns_packet_equal(const DnsPacket *a, const DnsPacket *b) {
         return dns_packet_compare_func(a, b) == 0;
 }
 
+int dns_packet_ede_rcode(DnsPacket *p, uint16_t *ret_ede_rcode) {
+        assert(p);
+        bool has_ede = false;
+        uint16_t ede_rcode = _DNS_EDNS_OPT_MAX_DEFINED;
+        const uint8_t *d;
+        size_t l;
+
```

Comment:
pleas also add `assert(ret_ede_rcode)`, since you strictly require the parameter to be passed.

(alternatively only set `*ret_ede_rcode` after validating via `if (ret_edec_rcode)` below that it is valid.

Reply:
Might as well not require it, so I'll check for ret_ede_rcode now.